modules/SofaGeneralEngine/src/SofaGeneralEngine/NearestPointROI.inl

```diff
@@ -75,11 +110,39 @@ void NearestPointROI<DataTypes>::reinit()
         msg_error() << "2 valid mechanicalobjects are required.";
         return;
     }
+
+    this->d_componentState.setValue(sofa::core::objectmodel::ComponentState::Valid);
     doUpdate();
 }
 
 template <class DataTypes>
 void NearestPointROI<DataTypes>::doUpdate()
+{
+    if (d_useRestPosition.getValue())
+    {
+        const VecCoord& x1 = this->mstate1->read(core::ConstVecCoordId::restPosition())->getValue();
+        const VecCoord& x2 = this->mstate2->read(core::ConstVecCoordId::restPosition())->getValue();
+
+        if (x1.size() == 0 || x2.size() == 0)
+            return;
+
+        computeNearestPointMaps(x1, x2);
+    }
+    else
+    {
+        const VecCoord& x1 = this->mstate1->read(core::ConstVecCoordId::position())->getValue();
+        const VecCoord& x2 = this->mstate2->read(core::ConstVecCoordId::position())->getValue();
+
+        if (x1.size() == 0 || x2.size() == 0)
+            return;
```

Comment:
you will have this warning at init for sure, because at init the mechanicalObject are not necessary yet filled. 
this method should be called the first time positions are changed but I know this method is called at init from the updateVisitor or something like this.

Moreover if one mechanicalObject is empty this means the output map will be empty and it could be totally right. You could imagine using this engine between and full mesh and new points accumulating in a second mechanicalObject